plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergSplitSource.java

```diff
@@ -304,6 +277,54 @@ private CompletableFuture<ConnectorSplitBatch> getNextBatchInternal(int maxSize)
         return completedFuture(new ConnectorSplitBatch(splits, isFinished()));
     }
 
+    private Iterator<FileScanTaskWithDomain> prepareFileTasksIterator(List<FileScanTaskWithDomain> fileScanTasks)
```

Comment:
you are materializing `FileScanTaskWithDomain` so just return `LIst< FileScanTaskWithDomain>` here

Reply:
It's more convenient to return iterator as the rest of the existing code is written to work on an iterator